### ISSUE TITLE
Add WithDisposableWarningSuppressedWhenDisposableNotOverridden(Type) method to Registration

### ISSUE THREAD

**Issue by dotnetjunkie** [simpleinjector/SimpleInjector#790]
Quite some framework base types, such as ASP.NET MVC's `Controller`, ASP.NET Core MVC's `Controller`, SignalR's `Hub` implement `IDisposable`, while implementing an empty `Dispose` method. This is a design flaw, because base types should hardly ever implement `IDisposable`. Still, framework builders keep adding those empty `Dispose` methods and this complicates integrating with Simple Injector, because Simple Injector does not dispose transient registrations.

To help its users, Simple Injector will detect any registered disposable transient and will throw an exception when `Verify()` is called. This is helpful, but still hard to fix for the user, especially because they themselves didn't override `Dispose`, and not disposing is not a problem.

Because of that, Simple Injector's ASP.NET and ASP.NET Core integration suppress this specific warning in case the user didn't override the protected `Dispose` method.

Although fixed for these technologies, new technologies (e.g. SignalR Core) keep popping up where the developers invalidly implement an empty `Dispose` method on their base type, and this is something we can't prevent (we tried, but didn't prevail). This means that Simple Injector should simplify this suppression, which is what this feature request is about.

My suggestion is to have a method (or extension method) on `Registration` that implements this. For instance:

``` c#
container.AddRegistration(typeof(SomeType),
    Lifestyle.Transient.CreateRegistration(typeof(SomeType), container)
        .WithDisposableWarningSuppressedWhenDisposableNotOverridden(typeof(BaseType)));
```

Problem, though, is that this method name is overly long; this is because it is very specific. It's hard, however, to make it more generic in nature.

Should this, however, be implemented in the core library, or in an integration project? If it is an integration project, the only likely place is the `SimpleInjector.Integration.ServiceCollection` package, but this might already be too specific.

**Comment by dotnetjunkie** [comment]
[feature-790](https://github.com/simpleinjector/SimpleInjector/tree/feature-790) branch created.

**Comment by dotnetjunkie** [comment]
A perhaps more natural, and simpler solution is to make this suppression global, part of the `ContainerOptions`:

``` c#
container.Options
    .SuppressDisposableTransientVerificationWhenDisposeIsNotOverriddenFrom(
        typeof(BaseType));
```

This simplifies registration (and even simplifies things for the official integration), and even removes the need to have a new `AddRegistrations` (see #789).

**Comment by dotnetjunkie** [comment]
A third option is to add a new lifestyle to Simple Injector that ensures disposal of those transient components. This, however, is tricky, and is a very different design philosophy. Not something we should take lightly.

**Comment by dotnetjunkie** [comment]
Another option that solves this problem is implemented in #793. Because it quite elegantly solves the problem, I don't see any need to add this quite verbose `WithDisposableWarningSuppressedWhenDisposableNotOverridden` method name for now. I therefore close this issue.
